pandas/core/dtypes/concat.py

```diff
@@ -559,3 +562,38 @@ def convert_sparse(x, axis):
         # coerce to object if needed
         result = result.astype('object')
     return result
+
+
+def _concat_indexes_same_dtype_rangeindex(indexes):
+
```

Comment:
can you add a comment about what this is doing / guarantees and example would be nice as well.